File: Greedy/python/10_저울.py
# fast campus 강의
# https://www.acmicpc.net/problem/2437
# 1

N, A = int(input()), sorted(list(map(int, input().split())))

# 만들 수 있는 최저
result = 0

for i in A:
    if i <= result + 1:
        result += i
    else:
        break

print(result + 1)

File: Greedy/c++/10_저울.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/2437
// 2

#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main()
{
	int N;
	cin >> N;

	vector<int> arr(N);
	for (int i = 0; i < N; i++)
		cin >> arr[i];
	sort(arr.begin(), arr.end());

	int result = 0;
	for (auto a : arr) {
		if (a <= result + 1)
			result += a;
		else
			break;
	}
	cout << result + 1;
}